staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -66,7 +66,7 @@ type Validator struct {
 // only errors.
 // Adding perCallLimit as input arg for testing purpose only. Callers should always use const PerCallLimit as input
 func NewValidator(s *schema.Structural, isResourceRoot bool, perCallLimit uint64) *Validator {
-	return validator(s, true, model.SchemaDeclType(s, isResourceRoot), perCallLimit)
+	return validator(s, isResourceRoot, model.SchemaDeclType(s, isResourceRoot), perCallLimit)
```

Comment:
do we have a test that would have caught this bug in isolation? should we add one?

Reply:
Yes, I was just thinking about the flow. I think a defaulter test that verifies that CEL expressions don't end up with spurious metadata fields available on a non-resource-root is how we should check this. I'll add shortly (have a 1hr meet starting momentarily).

Reply:
Oh wait, this doesn't test easily as easily as I had thought. It  turns out that the error was benign. That `isRootResource` field only pipes into a very specific path that the effect is not user visible (it ends up passing a structural schema with metadata fields declared to UnstructuredToVal), but the decls don't have the fields so they're not visible in CEL after type checking.